src/Aspire.Dashboard/Components/Controls/SpanDetails.razor

```diff
@@ -25,22 +25,61 @@
-                      NameSort="_nameSort"
-                      ValueSort="_valueSort"
-                      HighlightText="@_filter" />
-        <h4>Application</h4>
-        <PropertyGrid TItem="SpanPropertyViewModel"
-                      Items="@FilteredApplicationItems"
-                      GridTemplateColumns="1fr 2fr"
-                      NameColumnValue="(vm) => vm.Name"
-                      ValueColumnValue="(vm) => vm.Value"
-                      NameSort="_nameSort"
-                      ValueSort="_valueSort"
-                      HighlightText="@_filter" />
+        <FluentAccordion>
+            <FluentAccordionItem Heading="Span" Expanded="true">
+                <PropertyGrid TItem="SpanPropertyViewModel"
+                              Items="@FilteredItems"
+                              GridTemplateColumns="1fr 2fr"
+                              NameColumnValue="(vm) => vm.Name"
+                              ValueColumnValue="(vm) => vm.Value"
+                              NameSort="_nameSort"
+                              ValueSort="_valueSort"
+                              HighlightText="@_filter" />
+            </FluentAccordionItem>
+            <FluentAccordionItem Heading="Application">
+                <PropertyGrid TItem="SpanPropertyViewModel"
+                              Items="@FilteredApplicationItems"
+                              GridTemplateColumns="1fr 2fr"
+                              NameColumnValue="(vm) => vm.Name"
+                              ValueColumnValue="(vm) => vm.Value"
+                              NameSort="_nameSort"
+                              ValueSort="_valueSort"
+                              HighlightText="@_filter" />
+            </FluentAccordionItem>
+            <FluentAccordionItem Heading="Events">
+                @if (FilteredSpanEvents.Count() > 0)
+                {
+                    <div slot="end">
+                        <FluentBadge Appearance="Appearance.Accent"
+                                     Circular="true">
+                            @FilteredSpanEvents.Count()
+                        </FluentBadge>
+                    </div>
+                }
+                <PropertyGrid TItem="OtlpSpanEvent"
+                              Items="@FilteredSpanEvents"
+                              GridTemplateColumns="1fr 2fr"
+                              NameColumnValue="(vm) => DurationFormatter.FormatDuration(vm.TimeOffset(ViewModel.Span))"
+                              NameColumnTitle="@(Loc[nameof(ControlsStrings.TimeOffsetColumnHeader)])"
+                              ValueColumnValue="(vm) => vm.Name"
+                              ValueColumnTitle="@(Loc[nameof(ControlsStrings.EventColumnHeader)])"
+                              IsNameSortable="false"
+                              IsValueSortable="false"
+                              HighlightText="@_filter">
+                    <ExtraValueContent>
```

Comment:
Do you have a screenshot of what this looks like? I'm having trouble envisioning a property grid within a property grid value column given our information density already.

Reply:
![image](https://github.com/dotnet/aspire/assets/303201/3298d4c7-f645-4b90-99c1-d1074507c3ee)

Span events are rare. Events with attributes are rarer still (I had to add some fake data to get the UI to show up). I don't expect this UI to be shown much at all.

I think it is Good Enough until there is feedback otherwise.

Reply:
Added a header to make it clearer what the values are:

![image](https://github.com/dotnet/aspire/assets/303201/db2acad3-20d5-40e1-9400-39a81cf91d81)